### PR TITLE
[NativeAOT] Remove obsolete C++ name mangling

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

**Review by jkotas — APPROVED** [review]
Thanks

File: src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs [hunk 1]
```diff
@@ -69,7 +64,7 @@ public override string SanitizeName(string s, bool typeName = false)
                 sb ??= new StringBuilder(s, 0, i, s.Length);
 
                 // For CppCodeGen, replace "." (C# namespace separator) with "::" (C++ namespace separator)
-                if (typeName && c == '.' && _mangleForCplusPlus)
```
Comment (ThomasGoulet73) [inline src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs:72]: Shouldn't this whole if be removed ? In other places you remove code using the assumption that _mangleForCplusPlus is always false but now code in this if will be executed when it wasn't before.

(I don't have any context here so this change might be right. I just saw the change and I found it weird.)

Comment (MichalStrehovsky) [reply src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs:72]: Yes, would be nice to delete the comment above too and other references to C++ in this file

Comment (filipnavara) [reply src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs:72]: You are right. Thanks for noticing!

**Pull request merged**